camel-core/src/main/java/org/apache/camel/component/file/GenericFile.java

```diff
@@ -151,7 +165,7 @@ public void populateHeaders(GenericFileMessage<T> message) {
                 message.setHeader("CamelFileExtendedAttributes", extendedAttributes);
             }
             
-            if (probeContentType && file instanceof File) {
+            if (isProbeContentTypeFromEndpoint || (probeContentType && file instanceof File)) {
```

Comment:
I have my own review :) this line should be like 
`if ((isProbeContentTypeFromEndpoint || probeContentType) && file instanceof File)`